src/panels/lovelace/editor/hui-edit-card.ts

```diff
@@ -377,26 +452,27 @@ export class HuiEditCard extends hassLocalizeLitMixin(LitElement) {
   }
 
   private _isConfigChanged(): boolean {
+    if (this.viewId) {
+      return true;
+    }
     const configValue =
       this._configValue!.format === "yaml"
         ? yaml.safeDump(this._configValue!.value)
         : this._configValue!.value;
-    return JSON.stringify(configValue) !== JSON.stringify(this._originalConfig);
+    return JSON.stringify(configValue) !== JSON.stringify(this.cardConfig);
   }
 
-  private async _loadConfigElement(conf: LovelaceCardConfig): Promise<void> {
+  private async _loadConfigElement(conf: LovelaceCardConfig): Promise<boolean> {
```

Comment:
For a future PR, we should look if we can split out the editor piece from the editor dialog/data handling piece. A card editor can take a `CardConfig` and `PreferredUI (YAML/UI)`. All the async loading of elements and which editor to show can just be in that element. The dialog element can still show the toggle YAML button to control the editor, or that can be moved into the editor element too (under a … menu)